### PR TITLE
feature: Add some internal hidden options needed for reflect

### PR DESCRIPTION
**Pull request by grgbkr**
Adds
1. disableLicensing
2. disableMutationRecovery

### CODE REVIEW

File: src/replicache.ts [hunk 1]
```diff
@@ -279,6 +279,9 @@ export class Replicache<MD extends MutatorDefs = {}> {
   private _persistIsScheduled = false;
   private _recoveringMutations = false;
 
+  private readonly _disableLicensing: boolean = false;
```
Comment (phritz) [inline src/replicache.ts:282]: I don't know how your brains work but for mine "enableFoo" is so much easier to keep in mind than "disableFoo" which if on/true has the effect of turning something off.&lt;/shed&gt;

**Review by phritz — APPROVED** [review]
LGTM nice and simple!

File: src/replicache.ts (hunk 1 above)
Comment (grgbkr) [reply src/replicache.ts:282]: I don't feel strongly at all about this.  

My choice of disable vs enable is based on the default.  If the default behavior is enabled, then the option is to disable it. 

For me
```
new Replicache({
  disableLicensing: true
});
```

scans faster than

```
new Replicache({
  enableLicensing: false 
});
```

Comment (aboodman) [reply src/replicache.ts:282]: I don't care if it's internal API.

Comment (arv) [reply src/replicache.ts:282]: I'm also in favor of enable over disable in general.

Having default be true is not an issue to me

File: src/replicache.ts [hunk 2]
```diff
@@ -356,6 +359,17 @@ export class Replicache<MD extends MutatorDefs = {}> {
...
+      disableLicensing?: boolean;
+      disableMutationRecovery?: boolean;
+    };
+    if (internalOptions.disableLicensing) {
+      this._disableLicensing = true;
+    }
+    if (internalOptions.disableMutationRecovery) {
+      this._disableMutationRecovery = true;
+    }
+
```
Comment (arv) [inline src/replicache.ts:372]: I would do something more along the lines of:

```ts
type InternalOptions = {
  disableLicensing?: boolean;
  disableMutationRecovery?: boolean;
};
const {
  disableLicensing = false,
  disableMutationRecovery = false,
} = options as InternalOptions;
this._disableLicensing = disableLicensing;
this._disableMutationRecovery = disableMutationRecovery;
```

And then remove the initializer in the class field declarations.

File: src/replicache-mutation-recovery.test.ts [hunk 3]
```diff
@@ -1155,6 +1156,19 @@ test('mutation recovery returns early without running if push is disabled', asyn
   expect(await rep.recoverMutations()).to.equal(false);
 });
 
+test('disableMutationRecovery hidden option', async () => {
+  const rep = await replicacheForTestingNoDefaultURLs(
+    'mutation-recovery-startup',
+    {
+      pullURL: 'https://diff.com/pull',
+      disableMutationRecovery: true,
+    } as unknown as ReplicacheOptions<Record<string, never>>,
```
Comment (arv) [inline src/replicache-mutation-recovery.test.ts:1165]: Maybe change `replicacheForTestingNoDefaultURLs` to allow the internal options so the tests do not all have to do this casting dance.

**Review by arv — COMMENTED** [review]
Concept LGTM. Just a few nits

File: src/replicache.ts (hunk 2 above)
Comment (grgbkr) [reply src/replicache.ts:372]: done

File: src/replicache-mutation-recovery.test.ts (hunk 3 above)
Comment (grgbkr) [reply src/replicache-mutation-recovery.test.ts:1165]: done

**Comment by grgbkr** [comment]
@phritz @arv thanks for the reviews, some follow up improvements based on your comments in https://github.com/rocicorp/replicache/pull/927

**Pull request merged**